src/main/java/org/folio/circulation/rules/ApplyCondition.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.circulation.rules;
+
+import static io.vertx.core.MultiMap.caseInsensitiveMultiMap;
+import static org.folio.circulation.resources.AbstractCirculationRulesEngineResource.ITEM_TYPE_ID_NAME;
+import static org.folio.circulation.resources.AbstractCirculationRulesEngineResource.LOAN_TYPE_ID_NAME;
+import static org.folio.circulation.resources.AbstractCirculationRulesEngineResource.LOCATION_ID_NAME;
+import static org.folio.circulation.resources.AbstractCirculationRulesEngineResource.PATRON_TYPE_ID_NAME;
+
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.Location;
+import org.folio.circulation.domain.User;
+import org.folio.circulation.support.http.server.WebContext;
+
+import io.vertx.core.MultiMap;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.With;
+
+@Getter
+@ToString
+@AllArgsConstructor
+public final class ApplyCondition {
```

Comment:
What is an `apply condition`? To me, these values collectively are the parameters needed to execute the rules, is that the case?

```suggestion
public final class RulesExecutionParameters {
```